scripts/bundle-images.sh

```diff
@@ -0,0 +1,240 @@
+            sleep $interval_secs
+        fi
+    done
+}
+
+function bundle_images() {
+    local images=$1
+    local tarball=$2
+
+    echo "Bundling images into ${tarball}..."
+    docker save -o ${tarball} ${images}
+    if [[ $? -ne 0 ]]; then
+        echo "Error: Failed to bundle images into ${tarball}"
+        exit 1
+    fi
+}
+
+
+function ctrl_c() {
+        echo "Caught CTRL-C, exiting..."
+        exit 1
+}
+
+echo -e "\nVerifying provider Deployments are ready...\n"
+
+# Verify each provider we support has deployed so we can get the images used
+# across the deployments.
+for template in $(find ${TEMPLATES_DIR} -name 'provider.yaml');
+do
+    result=$(grep 'kind: .*Provider' ${template})
+    provider_yaml=$(grep "${result}" -A2 ${template})
+    provider_kind=$(echo -e "${provider_yaml}" | $YQ e '.kind' -)
+    provider_name=$(echo -e "${provider_yaml}" | $YQ e '.metadata.name' -)
+    provider_kind_tolower=$(echo ${provider_kind} | tr '[:upper:]' '[:lower:]')
+
+    if [[ $provider_name == "" ]]; then
+        echo "Error: Cannot determine provider Name from ${template}"
+        exit 1
+    fi
+
+    if [[ $provider_kind_tolower == "" ]]; then
+        echo "Error: Cannot determine provider Kind from ${template}"
+        exit 1
+    fi
+
+    # controlplane is a special case which needs a hyphen.
+    if [[ $provider_kind_tolower == "controlplane" ]]; then
+        provider_kind_tolower="control-plane"
+    fi
+
+    # coreprovider does not have a provider prefix.
+    if [[ $provider_kind_tolower == "coreprovider" ]]; then
+        label_value=$(echo ${provider_name})
+    else
+        label_value=$(echo $(echo ${provider_kind_tolower} | sed -e 's/provider//g')-${provider_name})
+    fi
+
+    wait_for_deploy_exist "$LABEL_KEY=$label_value"
+
+    echo "Verifying Deployment(s) with ${LABEL_KEY}=${label_value} label is condition=available..."
+
+    ${KUBECTL} wait --for condition=available --timeout=10m deploy -l ${LABEL_KEY}=${label_value} -n ${NAMESPACE}
+    if [[ $? -ne 0 ]]; then
+        echo "Error: Timed out waiting for available Deployment with ${LABEL_KEY}=${label_value} label"
+        exit 1
+    fi
+done
+
+
+# Now that we know everything is deployed and ready, we can get all of images by
+# execing into the KIND cluster.
+control_plane=$(${KUBECTL} get nodes --no-headers -o custom-columns=":metadata.name")
+if [[ $? -ne 0 ]] || [[ $control_plane == "" ]]; then
+    echo "Error: Cannot get control plane node"
+    exit 1
+fi
+
+echo -e "\nPulling images for HMC components...\n"
+
+for image in $(docker exec -it ${control_plane} crictl images | sed 1,1d | awk '{print $1":"$2}' | grep -v 'kindest');
+do
+    if [[ $image == "" ]]; then
+        echo "Error: Failed to get image from KIND cluster, image string should not be empty"
+        exit 1
+    fi
+
+    if [[ $image == "localhost/hmc/controller:latest" ]]; then
+        # Don't try to pull and package the local controller image.
+        continue
+    fi
```

Comment:
I got this bundled for some reason
```
   "RepoTags": [
      "hmc/controller:latest"
    ],
```
Plus I don't see an actual controller image in the bundle. Since we're not pulling it here we should have a separate case for it. Maybe I'm missing something?
All other images seem to work fine

Reply:
The `hmc/controller` image is supposed to be added from the `IMG` set in `make docker-build`, so by default the `IMG` environment var is `hmc/controller:latest`, because that's what the image gets built as.  In CI, we'd set the IMG tag to `build.Version` like we do in the CI for e2e testing and set that variable and that image would get bundled.  We could also just try setting that from the get-go outside of IMG I guess.

Reply:
Ok, so if you do this outside of CI you have to set it to some value with proper tag. 
I believe short note on that in dev.md will suffice.